File: OOP/1_class_object.cpp
#include<bits/stdc++.h>
using namespace std;

// class A{
//     public:
//     int value;
//     //constructor
//     A(int value){
//         this->value = value;
//         cout<<"Constructor Called"<<endl;
//     }

//     //method
//     void display(){
//         cout<<"Hello\n";
//     }

//     //destructor
//     ~A(){
//         cout<<"Desonstructor Called"<<endl;
//     }
// };
                    
// int main(){
//     A obj(8);
//     obj.display();
//     // obj.value = 5;
//     cout<<obj.value<<endl;
            
              
//     return 0;
// }








class A{
public:
    A(){
        printf("I am Constructor\n");
    }

    void display(){
        cout<<"I am Display Method\n";
    }

    ~A(){
        cout<<"I am Destructor\n";
    }

};

int main(){
    A obj;
    obj.display();

}

File: OOP/1_class_object.py

class A:
    value = 5 #class variable
    def __init__(self) -> None:
        print("I am Constructor")
    
    def display(self):
        print("I am method")

    def __del__(self):
        print("I am destructor")

    def __str__(self) -> str:
        return "I am ClasS A"




obj = A()
obj1 = A()
obj.display()
obj.value = 5

print(obj.__dict__)





print(obj.value)
print(A.value)






